rfcs/text/0000-application-monitoring-for-ga.md

```diff
@@ -0,0 +1,125 @@
+|------|---------|-------------|--------------|
+| Centralized configuration for health checks, custom metrics, alerts, and dashboards | [dcp-monitoring](https://github.com/HumanCellAtlas/dcp-monitoring) | [github+terraform](https://github.com/HumanCellAtlas/dcp-monitoring) | adopt as best practice |
+| Visualization layer that can plot metrics from any datasource | Grafana | [dev/integration/staging](https://metrics.dev.data.humancellatlas.org/), [prod](https://metrics.data.humancellatlas.org/) | adopt as best practice |
+| Metrics for AWS applications and managed services | [CloudWatch Metrics](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#Metric) | [console](https://console.aws.amazon.com/cloudwatch/home?region=us-east-1#metricsV2:) | adopt as best practice for AWS |
+| Metrics for GCP applications and managed services | [Stackdriver Metrics](https://cloud.google.com/monitoring/api/metrics) | [console](https://console.cloud.google.com/monitoring) | adopt as best practice for GCP |
+| Alerting based on CloudWatch Metrics | [CloudWatch Alarms](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#CloudWatchAlarms) | [github+terraform](https://github.com/HumanCellAtlas/dcp-monitoring) | adopt as best practice |
+| Metrics based on operational log indices | [Elasticsearch Metrics](http://docs.grafana.org/features/datasources/elasticsearch/) | [dev/integration/staging](https://logs.dev.data.humancellatlas.org/), [prod](https://logs.data.humancellatlas.org/) | use not recommended, but sometimes necessary |
+| Build statuses, system health, and availability reporting | [Status API](https://github.com/HumanCellAtlas/status-api) | [dev/integration/staging](https://status.dev.data.humancellatlas.org/), [prod](https://status.data.humancellatlas.org/) | use optional |
+| Public presentation of build statuses, system health, and availability reporting | [DCP Status Page](https://github.com/HumanCellAtlas/humancellatlas.github.io) | [https://humancellatlas.github.io/](https://humancellatlas.github.io/) | temporary until formal status page is built
+
+### Black-box monitoring
+
+Black-box monitoring attempts to test system behavior as a user would see it.
+
+To estimate and check this behavior, implement an HTTP health check endpoint for your service and configure a [Route53 health check](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/route-53-concepts.html#route-53-concepts-health-check) in the [dcp-monitoring](https://github.com/HumanCellAtlas/dcp-monitoring) repository.
+
+Your system's health check should aggregate the health status of its API and any infrastructure it depends on. For example, if your service is a HTTP REST API that holds state in a database, the health check endpoint would return a healthy status if the API was working and a basic query, such as `SELECT 1`, could be performed against the database and an unhealthy status otherwise.
```

Comment:
The prior guidance was that `is API up?` was good enough for a level 1 health check by GA, if I remember correctly. Is this formally changing the expectations of all components for this timeline?

Reply:
Yes, this is what I am proposing. There are few reasons that I think this is a good idea.

1. It will help us identify outages.
1. Currently, our availability numbers are upper bound estimates. We won't have a good idea of what our true availability numbers are if we do not have something that better approximates a good smoke test of the system.
1. I don't think teams have to go super deep here to get most of the benefit and can readily stop when they hit diminishing returns.